server/plugin/utils.go

```diff
@@ -350,21 +344,13 @@ func (p *Plugin) GetSubscriptionsForAccessibleChannelsOrProjects(subscriptionLis
 		return filteredSubscriptionList, nil
 	}
 
+	// For each subscription check if the user is a member of the MM channel where the subscription is created
 	for _, subscription := range subscriptionList {
-		// For each subscription on a project check if a user is an admin or member of the MM channel to return subscriptions
-		if projectDetails, isProjectLinked := p.IsProjectLinked(projectList, serializers.ProjectDetails{OrganizationName: subscription.OrganizationName, ProjectName: subscription.ProjectName}); isProjectLinked {
-			if projectDetails.IsAdmin {
+		for _, channel := range channels {
+			if subscription.ChannelID == channel.Id {
```

Comment:
can't we use the function which was created to check if the user is part of the channel or not?

Reply:
We should not use that here because that function makes an API call for each channel to check the user's membership and here many large numbers of subscriptions could be present and for each subscription's channelID making 1 API call will be a too expensive operation.